BridgeAppSDK/SBAGenericStepViewController.swift

```diff
@@ -0,0 +1,951 @@
+     */
+    static open func doesSupport(_ step: ORKStep) -> Bool {
+                
+        let supportedAnswerFormats: [ORKAnswerFormat.Type] = [ORKTextChoiceAnswerFormat.self,
+                                                              ORKTextAnswerFormat.self,
+                                                              ORKBooleanAnswerFormat.self,
+                                                              ORKNumericAnswerFormat.self]
+        
+        let supportedQuestionTypes: [ORKQuestionType] = [.decimal,
+                                                         .integer,
+                                                         .text,
+                                                         .singleChoice,
+                                                         .multipleChoice,
+                                                         .boolean]
+        
+        if let formStep = step as? SBAFormStepProtocol,
+            let formItems = formStep.formItems {
+            
+            for item in formItems {
+                if let answerFormat = item.answerFormat {
+                    
+                    let formatOkay = supportedAnswerFormats.contains { (type) -> Bool in
+                        type == type(of: answerFormat)
+                    }
+                    
+                    let typeOkay = supportedQuestionTypes.contains(answerFormat.questionType)
+                    
+                    if !(formatOkay && typeOkay) {
+                        return false
+                    }
+                }
+            }
+        }
+        return true
+    }
+    
+    
+    // MARK: Initializers
+    
+    override public init(step: ORKStep, result: ORKResult?) {
+        super.init(step: step)
+        commonInit(with: result)        
+    }
+    
+    override public init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit(with: nil)
+    }
+    
+    func commonInit(with result: ORKResult?) {
+        setupModel(with: result)
+    }
+    
+    
+    // MARK: View lifecycle
+    
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // register for keyboard notifications
+        NotificationCenter.default.addObserver(self, selector: #selector(self.keyboardNotification(notification:)), name: NSNotification.Name.UIKeyboardWillChangeFrame, object: nil)
+        
+        // show or hide our navigation bar
+        navigationController?.setNavigationBarHidden(!shouldShowNavigationBar(), animated: false)
+        
+        setupViews()
+    }
+        
+    override open func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        NotificationCenter.default.removeObserver(self)
```

Comment:
Sorry, missed this before. Could you please be specific about what notification is removed. Since this inherits from another class and b/c it is open, it can be seriously frustrating trying to track down why a notification isn't being received that you are expecting. Only to realize that one of the subclassing level was doing a general-purpose remove all. 

Thanks.